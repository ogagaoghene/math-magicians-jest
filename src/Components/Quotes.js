import React from 'react';
import '../css/quotes.css';

const Quotes = () => (
  <div className="grid__container">
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms.
      It is about understanding.
    </p>
    <br />
    <strong>- William Paul Thurston</strong>
  </div>
);

export default Quotes;
