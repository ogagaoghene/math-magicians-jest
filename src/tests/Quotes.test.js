import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quotes from '../Components/Quotes'

describe("Snapshot testing", () => {
  it("Check if Quotes component is rendered", () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  }); 

  it('renders quote', () => {
    const { getByText } = render(<Quotes />);
    expect( screen.getByText('Mathematics is not about numbers, equations, computations, or algorithms. It is about understanding.') ).toBeInTheDocument();
   });
})

