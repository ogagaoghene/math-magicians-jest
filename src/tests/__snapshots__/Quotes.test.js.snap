// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Check if Quotes component is rendered 1`] = `
<div
  className="grid__container"
>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms. It is about understanding.
  </p>
  <br />
  <strong>
    - William Paul Thurston
  </strong>
</div>
`;
